### PR TITLE
added recovery integration test

### PR DESCRIPTION
**Pull request by hackaugusto**

### CODE REVIEW

File: raiden/tests/integration/test_recovery.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    token_address = token_addresses[0]
+    node_state = views.state_from_app(app0)
+    payment_network_id = app0.raiden.default_registry.address
+    token_network_identifier = views.get_token_network_identifier_by_token_address(
+        node_state,
+        payment_network_id,
+        token_address,
+    )
+
+    # make a ferw transfers from app0 to app2
```
Comment (palango) [inline raiden/tests/integration/test_recovery.py]: s/ferw/few

File: raiden/network/transport/udp/udp_transport.py [hunk 2]
```diff
@@ -269,7 +269,9 @@ def start_health_check(self, recipient):
                 self.nat_keepalive_timeout,
                 self.nat_invitation_timeout,
                 ping_nonce,
-            ))
+            )
+            greenlet_healthcheck.name = f'Healtcheck for {pex(recipient)}'
```
Comment (palango) [inline raiden/network/transport/udp/udp_transport.py]: s/Healtcheck/Healthcheck

**Comment by hackaugusto** [comment]
bors retry

**Comment by hackaugusto** [comment]
the tests exposed a race condition, this should be fixed by https://github.com/raiden-network/raiden/pull/1592

**Pull request merged**